threediresultstyler/threediresultstyler_dockwidget.py

```diff
@@ -0,0 +1,259 @@
+        self.addSQLiteButton.clicked.connect(self.add_sqlite_connection)
+        self.addPDOKButton.clicked.connect(self.load_pdok)
+        self.highlightCourantButton.clicked.connect(self.highlight_courant)
+        self.removeHighlightButton.clicked.connect(self.remove_highlights)
+        
+        self.scenarioBox.installEventFilter(self)
+        self.scenarioBox.currentIndexChanged.connect(self.select_scenario)
+        
+        #self.courantThreshold
+
+    def eventFilter(self,target,event):
+        if target == self.scenarioBox and event.type() == QEvent.MouseButtonPress:
+            self.fill_scenario_selector()
+
+        return False
+
+    def closeEvent(self, event):
+        self.closingPlugin.emit()
+        event.accept()
+
+    def select_scenario(self):
+        self.selected_scenario_index = self.scenarioBox.currentIndex()
+        #self.iface.messageBar().pushMessage("Warning", "Selected scenario index {}".format(self.selected_scenario_index), level=Qgis.Warning)
+        
+
+    def fill_scenario_selector(self):
+        global_settings_layer = QgsProject.instance().mapLayersByName("v2_global_settings")
+
+        try:
+            for feature in global_settings_layer[0].getFeatures():
+                self.scenarios[feature["name"]] = feature["id"]
+            self.scenarioBox.addItems(list(self.scenarios.keys()))
+        except:
+            self.iface.messageBar().pushMessage("Warning", "Load 3Di model first", level=Qgis.Warning)
+
+    def set_project_crs(self, event):
+        global_settings_layer = QgsProject.instance().mapLayersByName("v2_global_settings")[0]
+        feature = list(global_settings_layer.getFeatures())[self.selected_scenario_index]
+        QgsProject.instance().setCrs(QgsCoordinateReferenceSystem(int(feature["epsg_code"])))
+
+    def load_dem(self, event):
+        global_settings_layer = QgsProject.instance().mapLayersByName("v2_global_settings")[0]
+        source = Path(global_settings_layer.dataProvider().dataSourceUri().split(' ')[0].split('=')[1][1:-1])
+        model_directory = source.parents[0]
+
+        root = QgsProject.instance().layerTreeRoot()
+        if not root.findGroup('background'):
+            grp_background = root.insertGroup(-1, "background")    
+        else:
+            grp_background = root.findGroup('background')
+            
+        feature = list(global_settings_layer.getFeatures())[self.selected_scenario_index]
+
+        dem_file = model_directory / Path(feature["dem_file"])
+
+        fileInfo = QFileInfo(str(dem_file))
+        path = fileInfo.filePath()
+        baseName = fileInfo.baseName()
+        
+        dem_layer = QgsRasterLayer(str(dem_file),'Digital Elevation Model')
+        QgsProject.instance().addMapLayer(dem_layer, False)
+        grp_background.insertChildNode(0,QgsLayerTreeLayer(dem_layer))
+        dem_layer.loadNamedStyle(os.path.join(os.path.dirname(__file__), 'styles\DEM.qml'))
```

Comment:
Die backslash werkt alleen op windows. Beter een slash doen.
En als je os.path.join doet is het nog netter om het zo te doen:
```suggestion
        dem_layer.loadNamedStyle(os.path.join(os.path.dirname(__file__), 'styles', 'DEM.qml'))
```